qa/rpc-tests/getblocktemplate.py

```diff
@@ -36,40 +50,59 @@ def setup_network(self, split=False):
+        # This part of the test should be improved, submit some V4 transactions
+        # and varying combinations of shielded and transparent
+        for _ in range(5):
+            # submit a tx with a shielded output
+            taddr0 = get_coinbase_address(node)
+            zaddr = node.z_getnewaddress('sapling')
+            recipients = [{"address": zaddr, "amount": Decimal('0.1')}]
+            myopid = node.z_sendmany(taddr0, recipients, 1, 0)
+            wait_and_assert_operationid_status(node, myopid)
+
+            # submit a tx with a transparent output
+            outputs = {node.getnewaddress():0.2}
+            node.sendmany('', outputs)
 
-        # Test 1: Default to coinbasetxn
-        tmpl = node.getblocktemplate()
-        assert('coinbasetxn' in tmpl)
-        assert('coinbasevalue' not in tmpl)
+        print("Getting block template (for height 2)")
+        gbt = node.getblocktemplate()
 
-        # Test 2: Get coinbasetxn if requested
-        tmpl = node.getblocktemplate({'capabilities': ['coinbasetxn']})
-        assert('coinbasetxn' in tmpl)
-        assert('coinbasevalue' not in tmpl)
+        prevhash = int(gbt['previousblockhash'], 16)
+        blockcommitmentshash = int(gbt['defaultroots']['blockcommitmentshash'], 16)
+        nTime = gbt['mintime']
+        nBits = int(gbt['bits'], 16)
 
-        # Test 3: coinbasevalue not supported if requested
-        tmpl = node.getblocktemplate({'capabilities': ['coinbasevalue']})
-        assert('coinbasetxn' in tmpl)
-        assert('coinbasevalue' not in tmpl)
+        f = BytesIO(hex_str_to_bytes(gbt['coinbasetxn']['data']))
+        coinbase = CTransaction()
+        coinbase.deserialize(f)
 
-        # Test 4: coinbasevalue not supported if both requested
-        tmpl = node.getblocktemplate({'capabilities': ['coinbasetxn', 'coinbasevalue']})
-        assert('coinbasetxn' in tmpl)
-        assert('coinbasevalue' not in tmpl)
+        print("Creating_block from template")
+        block = create_block(prevhash, coinbase, nTime, nBits, blockcommitmentshash)
 
-        # Test 5: General checks
-        tmpl = node.getblocktemplate()
-        assert_equal(16, len(tmpl['noncerange']))
-        # should be proposing height 2, since current tip is height 1
-        assert_equal(2, tmpl['height'])
+        # copy the non-coinbase transactions from the block template to the block
+        for gbt_tx in gbt['transactions']:
+            f = BytesIO(hex_str_to_bytes(gbt_tx['data']))
+            tx = CTransaction()
+            tx.deserialize(f)
+            block.vtx.append(tx)
+        # TODO: the merkle root calculated here doesn't match the merkle root
+        # in the block template (they should); the one in the template is
+        # correct because the block is accepted by submitblock. There is probably
+        # a bug in calc_merkle_root(), or this test is doing something wrong.
+        block.hashMerkleRoot = block.calc_merkle_root()
```

Comment:
My guess is that this hasn't been updated for ZIP 244 yet.

Reply:
This is now working!